packages/openapi3/test/tsp-openapi3/parameters.test.ts

```diff
@@ -0,0 +1,155 @@
+import { assert, describe, expect, it } from "vitest";
+import { tspForOpenAPI3 } from "./utils/tsp-for-openapi3.js";
+
+describe("converts top-level parameters", () => {
+  (["query", "header", "path"] as const).forEach((location) => {
+    it(`Supports location: ${location}`, async () => {
+      const serviceNamespace = await tspForOpenAPI3({
+        parameters: {
```

Comment:
Do we want to convert / check for pattern and other core constraints in parameters, or do we have such tests already elsewhere?

Reply:
Many of the constraints (pattern, format, mix/max value) are part of the `param-decorators` spec that's validated with the tests. Happy to add more specific tests here though too.

Reply:
Added more tests for constraints. I didn't add tests for the query/path arguments at this time - will do so as part of https://github.com/microsoft/typespec/issues/4298 (update transformations to work with the URI template changes)